File: sdk/node-ts/src/registry.ts
/** Discriminator tags for `RegistryAuth`. */
export type RegistryAuthKind = "anonymous" | "basic";

export const RegistryAuthKinds: readonly RegistryAuthKind[] = [
  "anonymous",
  "basic",
] as const;

/** Credentials for pulling private OCI images. */
export type RegistryAuth =
  | { kind: "anonymous" }
  | { kind: "basic"; username: string; password: string };

File: packages/microsandbox-types/rust/lib/registry.rs
//! Registry authentication contracts shared by local and cloud clients.

use serde::{Deserialize, Serialize};

//--------------------------------------------------------------------------------------------------
// Types
//--------------------------------------------------------------------------------------------------

/// Authentication credentials for OCI registry access.
#[derive(Debug, Clone, Default, Serialize, Deserialize)]
pub enum RegistryAuth {
    /// No authentication. Works for public registries.
    #[default]
    Anonymous,
    /// Username and password or token authentication.
    Basic {
        /// Registry username.
        username: String,
        /// Registry password or token.
        password: String,
    },
}
